libvirt/scripts/nested-virtualization/startup-script.sh

```diff
@@ -1,17 +1,44 @@
 #!/usr/bin/env bash
 
-apt update
-apt install -y \
-  uml-utilities \
-  qemu-kvm \
-  bridge-utils \
-  virtinst \
-  libvirt-daemon-system \
-  libvirt-clients
-
-virsh net-start default
-
-tunctl -t tap0
+# shellcheck source=/dev/null
+source "/etc/os-release"
+
+if [[ "$ID" == "rhel" || "$ID" == "centos" ]]
+then
+  # Use yum
+  yum install -y \
+    qemu-kvm \
+    libvirt \
+    libvirt-python \
+    libguestfs-tools \
+    virt-install
+fi
+
+if [[ "$ID" == "debian" || "$ID" == "ubuntu" ]]
+then
+  # Use apt
+  apt update
+  apt install -y \
+    uml-utilities \
+    qemu-kvm \
+    bridge-utils \
+    virtinst \
+    libvirt-daemon-system \
+    libvirt-clients
+fi
+
```

Comment:
It is really cool that you added those conditionals to play nicely with all those OSs 👍

Reply:
Thanks!